code/game/world.dm

```diff
@@ -131,12 +131,43 @@ GLOBAL_VAR_INIT(bypass_tgs_reboot, world.system_type == UNIX && world.byond_buil
 /world/Reboot(ping)
 	if(ping)
 		send2update(CONFIG_GET(string/restart_message))
-		send2update("Round ID [GLOB.round_id] finished | Next Ground Map: [SSmapping.next_map_configs[GROUND_MAP] ? SSmapping.next_map_configs[GROUND_MAP].map_name : SSmapping.configs[GROUND_MAP].map_name] | Next Ship Map: [SSmapping.next_map_configs[SHIP_MAP] ? SSmapping.next_map_configs[SHIP_MAP].map_name : SSmapping.configs[SHIP_MAP].map_name] | Round End State: [SSticker.mode?.round_finished] | Players: [length(GLOB.clients)]")
+		var/list/msg = list()
+
+		if(GLOB.round_id)
+			msg += "Round ID [GLOB.round_id] finished"
+
+		var/datum/map_config/next_map
+
+		if(length(SSmapping.next_map_configs))
```

Comment:
Now I am 100% sure this will runtime. Imagine only one map has the next config (since only one was voted for, the current case, meaning the length of the list will be 1, both will pass, but it will runtime with 1 case.

Reply:
Started up, changed the map immediately, populating the list to hold a single item:
![Xf2cqWq](https://user-images.githubusercontent.com/42041276/61326957-4567db00-a7ee-11e9-8f7c-3c5e53fd4508.png)
Then restarted. It didn't runtime, worked normally.

The runtime happened when the list was empty/null, because then `SSmapping.next_map_configs[GROUND_MAP]` became `SSmapping.null[GROUND_MAP]`
If you have a single item in the list, then you can look for string indexes that don't exist (just not integer ones, because then you are trying to access the list directly).

What happens instead is that `next_map = SSmapping.next_map_configs[SHIP_MAP]` returns a `null` value, because that index is empty. Which is fine because there's an `if(next_map)` check next.

Though now that you mention it there are two `if(length(SSmapping.next_map_configs))` checks, and those could be nested into a single one.

Reply:
I may be stupid then, carry on